pkg/agent/bgp/interface.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bgp
+
+import (
+	"antrea.io/antrea/pkg/apis/crd/v1alpha1"
+)
+
+// Interface defines the methods for managing a BGP (Border Gateway Protocol) process.
+// Currently, only the goBGP implementation is available.
+// More implementations might be added later.
+type Interface interface {
+	// Start initiates the BGP process.
+	Start() error
+
+	// Stop terminates the BGP process.
+	Stop() error
+
+	// AddPeer adds a new BGP peer.
+	AddPeer(peer PeerConfig) error
+
+	// UpdatePeer updates an existing BGP peer.
+	UpdatePeer(peer PeerConfig) error
+
+	// RemovePeer removes a specified BGP peer.
+	RemovePeer(peer PeerConfig) error
+
+	// GetPeers retrieves the current status of all BGP peers.
+	GetPeers() ([]PeerStatus, error)
+
+	// AdvertiseRoutes announces the specified routes to BGP peers for the given IP family.
+	AdvertiseRoutes(routes []Route) error
+
+	// WithdrawRoutes withdraws the specified routes from BGP peers for the given IP family.
+	WithdrawRoutes(routes []Route) error
+
+	// GetRoutes retrieves the advertised / received routes to / from the given peer for the given IP family.
+	GetRoutes(routeType RouteType, peerAddress string) ([]Route, error)
+}
+
+const (
+	// DefaultBGPListenPort is the default port for BGP server.
+	DefaultBGPListenPort int32 = 179
+
+	// DefaultBGPGracefulRestartTimeSeconds is the default time for BGP graceful restart in seconds.
+	DefaultBGPGracefulRestartTimeSeconds int32 = 120
```

Comment:
I don't think these 2 constants are used anywhere?

Reply:
Will remove them. They are not used in this PR.